backend/src/main/java/com/google/sps/servlets/AddParticipantServlet.java

```diff
@@ -0,0 +1,151 @@
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Find immediate match if possible
+    FindMatchQuery query = new FindMatchQuery();
+    Match match = query.findMatch(getParticipants(datastore), newParticipant);
+
+    // Match found, add to datastore, delete matched participants from datastore
+    if (match != null) {
+      addMatchToDatastore(match, datastore);
+      deleteSecondParticipantFromDatastore(match.getSecondParticipant(), datastore);
+    } else {
+      // Match not found, insert participant entity into datastore
+      Entity participantEntity = new Entity("Participant");
+      participantEntity.setProperty("ldap", ldap);
+      participantEntity.setProperty("timeAvailableUntil", timeAvailableUntil);
+      participantEntity.setProperty("timezone", timezone);
+      participantEntity.setProperty("duration", duration);
+      participantEntity.setProperty("timestamp", timestamp);
+      datastore.put(participantEntity);
+    }
+
+    // Redirect back to the HTML page
+    response.sendRedirect("/index.html");
+  }
+
+  /** Return list of current participants from datastore*/
+  private List<Participant> getParticipants(DatastoreService datastore) {
+
+    // Create and sort participant queries by time
+    Query query = new Query("Participant").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    // Convert list of entities to list of participants
+    List<Participant> participants = new ArrayList<Participant>();
+    for (Entity entity : results.asIterable()) {
+        long id = (long) entity.getKey().getId();
+        String ldap = (String) entity.getProperty("ldap");
+        long timeAvailableUntil = (long) entity.getProperty("timeAvailableUntil"); 
+        String timezone = (String) entity.getProperty("timezone"); 
+        int duration = (int) entity.getProperty("duration"); 
+        long timestamp = (long) entity.getProperty("timestamp");
+        Participant newParticipant = new Participant(id, ldap, timeAvailableUntil, timezone, duration, timestamp);
+        participants.add(newParticipant);
+    }
+    return participants;
+  }
+
+  /** Add Match pair to datastore */
+  private void addMatchToDatastore(Match match, DatastoreService datastore) {
+    // Set properties of entity
+    Entity matchEntity = new Entity("Match");
+    matchEntity.setProperty("firstParticipant", match.getFirstParticipant());
+    matchEntity.setProperty("secondParticipant", match.getSecondParticipant());
+    matchEntity.setProperty("duration", match.getDuration());
+    matchEntity.setProperty("timestamp", match.getTimestamp());
+
+    // Insert entity into datastore
+    datastore.put(matchEntity);
+  }
+
+  /** Delete matched participants from datastore */
+  private void deleteSecondParticipantFromDatastore(Participant secondParticipant, DatastoreService datastore) {
```

Comment:
s/secondParticipant/participant

Where this method is used, secondParticipant is the variable used but this method doesn't know anything about first vs second